crates/swc/src/config/mod.rs

```diff
@@ -240,37 +244,106 @@ impl Options {
     }
 }
 
-/// Configuration related to source map generated by swc.
-#[derive(Clone, Serialize, Deserialize, Debug)]
-#[serde(untagged)]
+/// Configuration related to source maps generation.
+#[derive(Debug, Clone, Serialize, Eq, PartialEq)]
 pub enum SourceMapsConfig {
     Bool(bool),
-    Str(String),
+    Inline,
+    Linked,
```

Comment:
I think this should be `Linked(PathBuf)` instead

Reply:
`Linked` + `None` for `source_map_path` is an invalid state

Reply:
Yes, I also thought about this. I want to emphasize that this will not solve the dependent-type problem.

For example, the instantiation creates an invalid Path to a nonexistent source maps file already

```json
{
    "sourceMaps": "linked"
}
```

```rust
SourceMapsConfig::Linked(PathBuf::new())
```

The value of the `SourceMapsConfig::Linked` enum can not be universal. Its inner `PathBuf` value (which is the path to the source maps file) is dynamic and changes with every input.

Reply:
I tried this out and the code is even more ugly now.

Reply:
To get this right, the `SourceMapsConfig::Bool(bool)` needs to be replaced with `SourceMapsConfig::Enabled(PathBuf)` and `SourceMapsConfig::Disabled` respectively. Please let me know if I this is alright.

Reply:
Why do we need pathbuf as an argument? `true` simply generates `.map` file with same file name, so I think we should do same thing for `linked`, too

Reply:
The reason is that I would prefer to move entire generation of the .map path into the swc core crate. If we would do that, then we could also return a map path from there.

Reply:
I implemented it as you suggested, but I had to delete the fixture test for the linked option.

Reply:
I implemented it as you suggested, but I had to delete the fixture test for the linked option. This is the down-side of your idea.